Document how to connect to ipdb with docker exec

mstriemer opened the pull request:
I had originally removed the documentation on how to use `docker-utils` but that command is probably easier to remember so I left it in.

r? @andymckay

muffinresearch commented on docs/topics/hacking/debugging.rst:
```diff
@@ -15,8 +15,19 @@ As with ipdb normally just add a line in your code at the relevant point:
-installed docker-utils_ first)::
+Next open a shell on the running web container::
+
+    docker exec -t -i olympia_web_1 supervisorctl fg olympia
```
Gah that's is definitely not memorable, but at least it takes you straight to the foregrounded runserver.

muffinresearch replied:
This is exactly why docker-utils exists, perhaps this should be the verbose alternative to the former?

muffinresearch commented:
r+wc lgtm, nice to have the directness.

andymckay commented:
I have no attachment to docker-utils, I like it because its shorter, but increasing the amount of things people think they have to use isn't good. 

Could you also remove it from index.rst? https://github.com/mozilla/olympia/search?utf8=%E2%9C%93&q=docker-utils, if you do that r+ from me.

muffinresearch commented:
> Could you also remove it from index.rst? https://github.com/mozilla/olympia/search?utf8=%E2%9C%93&q=docker-utils, if you do that r+ from me.

If you remove the references to it you'll need to add in docs for how to get a shell with exec. However, I feel that forcing people to need to remember horribly verbose docker exec args is a backwards step. Is there any alternative.

mstriemer commented:
I added some commands to the Makefile `make shell` and `make debug`. I also updated `make initialize_docker` and `make update_docker` to run the commands in docker directly. This is making me think we should add a script like `./run-docker` that can do things like `./run-docker py.test path/to/my/test.py` and these commands can use that and I don't have to keep typing `docker-compose run web -- py.test path/to/my/test.py` each time I want to run the tests.

EnTeQuAk commented:
@mstriemer I like the new makefile actions, not a fan of a `run-docker` script though. Why not just put all this into the makefile so this is the only one thing we'd ever need for regular stuff?

muffinresearch commented:
>  I don't have to keep typing docker-compose run web -- py.test path/to/my/test.py each time I want to run the tests.

You shouldn't be needing to do that anyway. This is what docker exec is for, to allow commands to be run on the container without spinning up one just to test.  

It's also why docker-utils is useful because you can just drop into a shell and run test commands like you used to for local dev.

muffinresearch commented:
Fwiw I'm looking into making docker-utils take pluggable commands. E.g. useful commands that utils should run can be just plugged-in (e.g. scripts that live in a docker-utils dir or something).

As `docker-utils` wraps `docker-compose` already this makes a nice way to extend compose. It's similar to how github's hub command works e.g. the hub command adds sugar to `git` - and the intention is you alias it. If we get that working nicely then it would then be maybe worth suggesting it to upstream as a feature to `docker-compose` itself.

mstriemer commented:
Yeah, there's really no need for a script. If I want to run custom one-off commands I can alias that.

I like to run my tests in `vim` hence why I run them with `docker-compose run web`. It takes less than a second to boot the container which is pretty negligible compared to olympia's test suite (even for a single test) unfortunately. That being said I'll definitely alias the `docker exec` command and save that time from now on.

I'll merge this when I get back from lunch unless there are any concerns with the changes to the Makefile.

mstriemer commented:
As far as adding a test command to the Makefile: that seems like a good idea but I've never found `make` any good for accepting arguments so it would be something like `make test TEST=path/to/tests.py` unless there's a better way I don't know of.

EnTeQuAk commented:
@mstriemer I think the default workaround (at least one I know of) for accepting arguments would be doing something like

EnTeQuAk commented:
@mstriemer I think a proper workaround for accepting arguments in a makefile could be (at least the one I know of)

```
test: 
    docker-utils ... ${ARGS}
```

And use something like ´ARGS="-m 'not slowtest'"` or so. I think that could work.

mstriemer commented:
I added a `make make` command that will run a make command in docker.

`make make ARGS='test ARGS=apps/addons/tests/test_views.py'`

EnTeQuAk commented:
I like the changes and it seems to work great. r+ from my side

muffinresearch commented:
> make make ARGS='test ARGS=apps/addons/tests/test_views.py'

![mal-uh-whut](https://cloud.githubusercontent.com/assets/1514/10673605/9c2ee3d0-78ee-11e5-9a6a-1241bc0e4fa4.gif)
